checks/permissions.go

```diff
@@ -59,20 +60,18 @@ func validateMapPermissions(values map[interface{}]interface{}, path string,
 
 	// Iterate over the permission, verify keys and values are strings.
 	for k, v := range values {
-		switch key := k.(type) {
-		// String type.
-		case string:
-			if r, err = validatePermission(key, v, path, logf); err != nil {
-				return false, err
-			}
-
-			if !r {
-				permissionRead = false
-			}
-		// Invalid type.
-		default:
+		key, ok := k.(string)
+		if !ok {
 			return false, ErrInvalidGitHubWorkflowFile
 		}
+
+		if r, err = validatePermission(key, v, path, logf); err != nil {
+			return false, err
+		}
+
+		if !r {
```

Comment:
Shouldn't this be `else if`?

Reply:
I want to be able to log all offending lines. I think it helps users to fix everything at once. Otherwise, I'm worried some users will fix the one line that scorecard reports, and then realize they should fix the rest.